python_modules/libraries/dagster-dbt/dagster_dbt/components/dbt_project/component.py

```diff
@@ -240,37 +255,62 @@ def get_asset_check_spec(
         return self._base_translator.get_asset_check_spec(asset_spec, manifest, unique_id, project)
 
     @cached_property
-    def cli_resource(self):
-        return DbtCliResource(self.project)
+    def dbt_project(self) -> DbtProject:
+        if isinstance(self.project, DbtProject):
+            return self.project
```

Comment:
any concern this setup will break existing users? How much did you explore having `RemoteDbtProjectComponent` and shuffling stuff in to shared base class?